packages/smooth_app/lib/cards/product_cards/product_title_card.dart

```diff
@@ -22,22 +22,130 @@ class ProductTitleCard extends StatelessWidget {
   final bool isSelectable;
   final bool isRemovable;
 
+  @override
+  Widget build(BuildContext context) {
+    return Provider<Product>.value(
+      value: product,
+      child: Align(
+        alignment: AlignmentDirectional.topStart,
+        child: InkWell(
+          onTap: _hasProductName
+              ? () async {
+                  await Navigator.push<Product?>(
+                    context,
+                    MaterialPageRoute<Product>(
+                      builder: (BuildContext context) =>
+                          AddBasicDetailsPage(product),
+                    ),
+                  );
+                }
+              : null,
```

Comment:
Shouldn't that be the other way around

Reply:
That is to say?

Reply:
I guess @M123-dev meant: "if the product name is NOT populated, a `tap` would open the `AddBasicDetailsPage`".
Like it was before (more or less):
```dart
onTap: (getProductName(product, appLocalizations) ==
                appLocalizations.unknownProductName)
            ? () async {
```
The way you coded here is the other way around.